Panic when attempting to refer to an external schema

My API is returning a JSON schema, and I'm trying to tell my OpenAPI clients that this is the case by referring to the well-known schema URL `https://json-schema.org/draft/2020-12/schema`.

I just want my `openapi.json` to contain something like:
```json
"responses": {
  "200": {
     "content": {
       "application/json": {
         "schema": {
           "$ref": "https://json-schema.org/draft/2020-12/schema"
         }
       }
     },
   },
```

I've attempted to do that like this (full example [here](https://go.dev/play/p/1MwuQitKwOS)):
```go
Responses: map[string]*huma.Response{
    "200": {
        Content: map[string]*huma.MediaType{
            "application/json": {
                Schema: &huma.Schema{
                    Ref: "https://json-schema.org/draft/2020-12/schema",
                },
            },
        },
    },
},
```

But this produces a panic:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x5c353c]

goroutine 1 [running]:
github.com/danielgtaylor/huma/v2.(*SchemaLinkTransformer).addSchemaField(0xc00011ae70, 0xc00012e210, 0xc00011aff0)
	/tmp/gopath2575097088/pkg/mod/github.com/danielgtaylor/huma/v2@v2.27.0/transforms.go:57 +0x7c
github.com/danielgtaylor/huma/v2.(*SchemaLinkTransformer).OnAddOperation(0xc00011ae70, 0xc00012e210, 0xc00015f2c0)
	/tmp/gopath2575097088/pkg/mod/github.com/danielgtaylor/huma/v2@v2.27.0/transforms.go:103 +0x23d
github.com/danielgtaylor/huma/v2.(*OpenAPI).AddOperation(0xc00012e210, 0xc00015f2c0)
	/tmp/gopath2575097088/pkg/mod/github.com/danielgtaylor/huma/v2@v2.27.0/openapi.go:1533 +0x306
github.com/danielgtaylor/huma/v2.Register[...]({_, _}, {{0x64ae2c, 0x3}, {0x64ad38, 0x1}, 0x0, 0x0, 0x0, {0x0, ...}, ...}, ...)
	/tmp/gopath2575097088/pkg/mod/github.com/danielgtaylor/huma/v2@v2.27.0/huma.go:893 +0x2405
main.main()
	/tmp/sandbox4189181463/prog.go:14 +0x2f6
```

This occurs because the `schema` variable returned [here](https://github.com/danielgtaylor/huma/blob/44aab0b99652b9da1aef125410bb2556a2cd68b3/transforms.go#L56) isn't `nil`-checked before it is accessed on the following line.

Is this the correct way to accomplish this and this is a bug? Or am I doing something wrong?

--- a/huma_test.go
+++ b/huma_test.go
@@ -2145,6 +2145,45 @@ Content-Type: text/plain
 			},
 		},
 		{
+			Name: "response-external-schema",
+			Register: func(t *testing.T, api huma.API) {
+				type Resp struct {
+					Body struct {
+						Greeting string `json:"greeting"`
+					}
+				}
+
+				huma.Register(api, huma.Operation{
+					Method: http.MethodGet,
+					Path:   "/response",
+					Responses: map[string]*huma.Response{
+						"200": {
+							Description: "Success",
+							Content: map[string]*huma.MediaType{
+								"application/json": {
+									Schema: &huma.Schema{
+										// Using an external schema should not break.
+										// https://github.com/danielgtaylor/huma/issues/703
+										Ref: "http://example.com/schemas/foo.json",
+									},
+								},
+							},
+						},
+					},
+				}, func(ctx context.Context, input *struct{}) (*Resp, error) {
+					resp := &Resp{}
+					resp.Body.Greeting = "Hello, world!"
+					return resp, nil
+				})
+			},
+			Method: http.MethodGet,
+			URL:    "/response",
+			Assert: func(t *testing.T, resp *httptest.ResponseRecorder) {
+				assert.Equal(t, http.StatusOK, resp.Code)
+				assert.JSONEq(t, `{"greeting":"Hello, world!"}`, resp.Body.String())
+			},
+		},
+		{
 			// Simulate a request with a body that came from another call, which
 			// includes the `$schema` field. It should be allowed to be passed
 			// to the new operation as input without modification.

--- a/transforms.go
+++ b/transforms.go
@@ -6,6 +6,7 @@ import (
 	"os"
 	"path"
 	"reflect"
+	"strings"
 )
 
 type schemaField struct {
@@ -49,7 +50,7 @@ func NewSchemaLinkTransformer(prefix, schemasPath string) *SchemaLinkTransformer
 }
 
 func (t *SchemaLinkTransformer) addSchemaField(oapi *OpenAPI, content *MediaType) bool {
-	if content == nil || content.Schema == nil || content.Schema.Ref == "" {
+	if content == nil || content.Schema == nil || content.Schema.Ref == "" || !strings.HasPrefix(content.Schema.Ref, "#/") {
 		return true
 	}
 

